### DISCUSSION TITLE
Vercel Deploy Button for custom site

### DISCUSSION THREAD

**Discussion by cosbgn** [vercel/community#1038] [Idea]
I have a website builder, but instead of hosting everything myself I would like for my customers to be able to host the site on Vercel. Currently I can:

- Generate a .zip with a static site
- Instruct the user on how to upload it to vercel

What I would like is to simply have a `deploy to vercel` button which instead of a git repository takes a url of a `zip file`. Is there any way I could do something like that?

**Comment by amyegan** [comment]
Nice idea @cosbgn! The [Deploy Button](https://vercel.com/docs/deploy-button) requires a public Git repository URL from GitHub, GitLab, or Bitbucket. 

You could work around this by temporarily creating a public repo for your customers to deploy from, assuming the site code is safe to put in a public repo. I believe you could create the repo with [GitHub API](https://docs.github.com/en/rest/repos/contents#create-a-file), [GitLab API](https://docs.gitlab.com/ee/api/projects.html#create-project), or [Bitbucket API](https://developer.atlassian.com/cloud/bitbucket/rest/api-group-projects/#api-workspaces-workspace-projects-post) and then put the Deploy Button for it on your site to make things easier for your customers. I haven't tried it myself, so that's just a hypothesis.

I'm adding your idea to the feature suggestions list.
